src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/DocumentProcessor.java

```diff
@@ -33,14 +35,18 @@ public void uploadPdfFiles(List<Pdf> pdfs, List<ScannableItem> scannedItems) {
 
         log.info("Document service response with file name and doc url {}", response);
 
-        scannedItems.forEach(item -> {
-            if (response.containsKey(item.getFileName())) {
-                item.setDocumentUrl(response.get(item.getFileName()));
-            } else {
-                throw new DocumentNotFoundException("Document metadata not found for file " + item.getFileName());
-            }
-        });
-
-        scannableItemRepository.saveAll(scannedItems);
+        List<String> filesWithoutUrl =
```

Comment:
I personally would put this in a method to simplify this method a bit and have a clear responsibility